python/setup.py

```diff
@@ -227,7 +227,7 @@ def get_packages(self):
 if setup_spec.type == SetupType.RAY:
     pandas_dep = "pandas >= 1.3"
     numpy_dep = "numpy >= 1.20"
-    pyarrow_dep = "pyarrow >= 6.0.1"
+    pyarrow_dep = "pyarrow!=16.0.1,>=6.0.1"
```

Comment:
the version that you mentioned that has issue seems to be 16.1.0

there is no 16.0.1 of pyarrow on pypi yet.